### PR TITLE
moved profile and editprofile to feature module

### PR DESCRIPTION
**Pull request by AdityaKumdale**
## Issue Fix

Jira Task: [MW-52](https://mifosforge.jira.com/browse/MW-52)

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

[MW-52]: https://mifosforge.jira.com/browse/MW-52?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature/profile/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+plugins {
+    alias(libs.plugins.mifospay.android.feature)
+    alias(libs.plugins.mifospay.android.library.compose)
+}
+
+android {
+    namespace = "org.mifospay.profile"
```
Comment (therajanmaurya) [inline feature/profile/build.gradle.kts]: rename package name to "org.mifospay.feature.profile"

File: feature/profile/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+dependencies {
+    implementation(projects.core.data)
+    implementation(libs.squareup.okhttp)
```
Comment (therajanmaurya) [inline feature/profile/build.gradle.kts:12]: why do you need okhttp here?

File: feature/profile/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    implementation(libs.compose.country.code.picker)
+    implementation(libs.compose.material)
+    implementation(libs.coil.kt.compose)
+    implementation(libs.androidx.appcompat)
```
Comment (therajanmaurya) [inline feature/profile/build.gradle.kts]: why ?

File: feature/profile/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    implementation(projects.mifospay)
```
Comment (therajanmaurya) [inline feature/profile/build.gradle.kts:16]: remove this, mifospay is parent module and this profile is child module.

File: feature/profile/src/main/kotlin/org/mifospay/feature/profile/ProfileItemCard.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+@Composable
+fun DetailItem(label: String, value: String) {
+    Text(
+        text = "$label: $value",
+        style = MaterialTheme.typography.bodyLarge,
+        modifier = Modifier.padding(bottom = 12.dp)
+    )
+}
```
Comment (therajanmaurya) [inline feature/profile/src/main/kotlin/org/mifospay/feature/profile/ProfileItemCard.kt:71]: why this does not have preview?

File: feature/profile/src/main/res/drawable/ic_contact.xml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (therajanmaurya) [inline feature/profile/src/main/res/drawable/ic_contact.xml]: Use MifosIcons instead vector drawable.

File: feature/profile/src/main/res/drawable/ic_camera.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (therajanmaurya) [inline feature/profile/src/main/res/drawable/ic_camera.xml]: Use MifosIcons instead vector drawable.

File: feature/profile/src/main/res/drawable/ic_bank.xml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (therajanmaurya) [inline feature/profile/src/main/res/drawable/ic_bank.xml]: Use MifosIcons instead vector drawable.

File: feature/profile/src/main/res/drawable/ic_setting.xml [hunk 9]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (therajanmaurya) [inline feature/profile/src/main/res/drawable/ic_setting.xml]: Use MifosIcons instead vector drawable.

File: feature/profile/src/main/res/drawable/qrcode_black.xml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (therajanmaurya) [inline feature/profile/src/main/res/drawable/qrcode_black.xml]: Use MifosIcons instead vector drawable.

File: feature/profile/src/main/res/values/colors.xml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (therajanmaurya) [inline feature/profile/src/main/res/values/colors.xml]: We have Colors.kt in designsystem module move these their or check it is already present to not.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: feature/profile/build.gradle.kts (hunk 2 above)
Comment (AdityaKumdale) [reply feature/profile/build.gradle.kts:12]: We need it for ResponseBody and inputStream : private fun convertResponseToBitmap(responseBody: ResponseBody?): Bitmap? {
        return try {
            responseBody?.byteStream()?.use { inputStream ->
                BitmapFactory.decodeStream(inputStream)
            }

File: feature/profile/build.gradle.kts (hunk 3 above)
Comment (AdityaKumdale) [reply feature/profile/build.gradle.kts]: We need these (despite profile being child module) : -
1) implementation(libs.compose.country.code.picker) -> for togi countrypicker implementation
2)implementation(libs.coil.kt.compose) -> AsyncImage implementation
3)implementation(libs.compose.material) -> can't use material  3 in togi conuntrypicker for ' TextFieldDefaults'

File: feature/profile/build.gradle.kts (hunk 4 above)
Comment (AdityaKumdale) [reply feature/profile/build.gradle.kts:16]: we need  'implementation(projects.mifospay)'  for BuildConfig in :-   val uri = FileProvider.getUriForFile(
        Objects.requireNonNull(context),
        BuildConfig.APPLICATION_ID + ".provider", file
    )

Comment (therajanmaurya) [reply feature/profile/build.gradle.kts:16]: just do hard coded package name for now

**Pull request merged**